KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/KeywordSearchIngestModule.java

```diff
@@ -515,7 +518,11 @@ private void postIndexSummary() {
         private boolean extractTextAndIndex(AbstractFile aFile, Map<String, String> extractedMetadata) throws IngesterException {
             ImageConfig imageConfig = new ImageConfig();
             imageConfig.setOCREnabled(KeywordSearchSettings.getOcrOption());
-            ProcessTerminator terminator = () -> context.fileIngestIsCancelled();
+            
+            List<ProcessTerminator> terminators = new ArrayList<>();
+            terminators.add(() -> context.fileIngestIsCancelled()); 
+            terminators.add(new TimedProcessTerminator(OCR_TIMEOUT_SECONDS));
+            HybridTerminator terminator = new HybridTerminator(terminators);
             Lookup extractionContext = Lookups.fixed(imageConfig, terminator);
```

Comment:
This will actually cancel any extractor that is listening to the terminator, instead of just the OCR task.
Could you instead add the TimedProcessTerminator (and the 30 minute constant) to the ImageConfig class instead?
You could then construct a HybridTerminator before the OCR task is started in TikaTextExtractor.